k8s/upgrade/src/bin/upgrade-job/events/event_recorder.rs

```diff
@@ -251,12 +251,17 @@ impl EventRecorder {
     }
 
     /// This method is intended for use when upgrade fails.
-    pub(crate) async fn publish_unrecoverable<Error>(&self, err: &Error)
+    pub(crate) async fn publish_unrecoverable<Error>(&self, err: &Error, validation_error: bool)
     where
         Error: Display,
     {
+        let action = if validation_error {
+            "Validation Failed"
```

Comment:
this should be an enum which both the job and the plugin can can use for checking what kind of error it is.

Reply:
done